lib/graphql/query/serial_execution/selection_resolution.rb

```diff
@@ -6,12 +6,19 @@ def self.resolve(target, current_type, selection, query_ctx)
           selection_result = {}
 
           selection.each_selection(type: current_type) do |name, subselection|
-            selection_result.merge!(query_ctx.execution_strategy.field_resolution.new(
+            field_result = query_ctx.execution_strategy.field_resolution.new(
               subselection,
               current_type,
               target,
               query_ctx
-            ).result)
+            ).result
+
+            if field_result.values[0] == GraphQL::Execution::Execute::PROPAGATE_NULL
```

Comment:
Actually, doing the same thing in get_finished_value won't really allow the null to propagate for objects, since at that point only the object has resolved and not its fields.  It is probably easiest to just stick with exceptions, especially if you are just going to pass exceptions to the type_error callback.